### PR TITLE
Add support for gesture reactions constraint

### LINKED ISSUE #118: Should web applications be aware of reaction effects added by OS to camera feeds?
See https://bugs.webkit.org/show_bug.cgi?id=262233 for background.
It would seem useful to at least allow web applications to know that such effects are being enabled, similarly to other effects like background blur.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes #118 (preliminary, for discussion)

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/mediacapture-extensions/pull/141.html" title="Last updated on Mar 28, 2024, 3:41 PM UTC (3c570ef)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-extensions/141/2fa3a6b...jan-ivar:3c570ef.html" title="Last updated on Mar 28, 2024, 3:41 PM UTC (3c570ef)">Diff</a>

### CODE REVIEW

**Comment by eladalon1983** [comment]
Rather than `reactions`, this should be in the vein of `heuristicReactions` or `heuristicEffects`.

The problem we seek to solve is that gesture-driven reactions misfire and effects are injected in contravention of the user's actual intentions. Having a `heuristicEffects` control would help prevent that.

In contrast, if we specified `reactions`, that would allow an app to suppress the user's explicitly delivered intentions (e.g. via menu interaction).

**Comment by jan-ivar** [comment]
AVCaptureDevice (`canPerformReactionEffects`) appears to call them "reaction effects", i.e. effects in response to user reactions.

IOW, the word "reaction" refers to the "user reaction" that _triggers_ the effect, not the effect itself.

Thinking of it this way, the word heuristic seems captured by reaction. E.g.

reactionEffect = heuristic effect
vs.
explicitlyTriggeredEffect = non-heuristic triggering of the same effect through clicking on this menu:  

So maybe `track.getSettings().reactionEffects`?

File: index.html [hunk 1]
```diff
@@ -1056,6 +1056,35 @@ <h2>Exposing MediaStreamTrack source background blur support</h2>
...
+         Web applications may either want to control or at least be aware that these heuristics are active and might trigger these effects at the source level.
+         This can for instance allow the web application to update its UI or to turn off these heuristics where having such effects tiggered accidentally might be considered insensitive or inappropriate.
+         For that reason, we extend {{MediaStreamTrack}} with the following properties.
+      </p>
+    </div>
+    <div>
+      <p>The WebIDL changes are the following:
+      <pre class="idl">
+partial dictionary MediaTrackSupportedConstraints {
+  boolean reactions = true;
```
Comment (jan-ivar) [inline index.html:1075]: Editors liked gesture better than heuristic. How about:
```suggestion
  boolean gestureReactions = true;
```

File: index.html [hunk 2]
```diff
@@ -1056,6 +1056,35 @@ <h2>Exposing MediaStreamTrack source background blur support</h2>
...
+};
+
+partial dictionary MediaTrackConstraintSet {
+  ConstrainBoolean reactions;
```
Comment (jan-ivar) [inline index.html:1079]:
```suggestion
  ConstrainBoolean gestureReactions;
```

File: index.html [hunk 3]
```diff
@@ -1056,6 +1056,35 @@ <h2>Exposing MediaStreamTrack source background blur support</h2>
...
+};
+
+partial dictionary MediaTrackSettings {
+  boolean reactions;
```
Comment (jan-ivar) [inline index.html:1083]:
```suggestion
  boolean gestureReactions;
```

File: index.html [hunk 4]
```diff
@@ -1056,6 +1056,35 @@ <h2>Exposing MediaStreamTrack source background blur support</h2>
...
+};
+
+partial dictionary MediaTrackCapabilities {
+  sequence&lt;boolean&gt; reactions;
```
Comment (jan-ivar) [inline index.html:1087]:
```suggestion
  sequence&lt;boolean&gt; gestureReactions;
```

**Comment by jan-ivar** [comment]
@eladalon1983 WDYT?

**Comment by eladalon1983** [comment]
LGTM

**Pull request merged**